src/types/any.rs

```diff
@@ -164,35 +174,40 @@ impl PyAny {
     /// ```
     pub fn setattr<N, V>(&self, attr_name: N, value: V) -> PyResult<()>
     where
-        N: ToPyObject,
+        N: IntoPy<Py<PyString>>,
         V: ToPyObject,
     {
         let py = self.py();
+
+        let attr_name = attr_name.into_py(py).into_ptr();
+        let value = value.to_object(py).into_ptr();
+
         unsafe {
-            err::error_on_minusone(
-                py,
-                ffi::PyObject_SetAttr(
-                    self.as_ptr(),
-                    attr_name.to_object(py).as_ptr(),
-                    value.to_object(py).as_ptr(),
-                ),
-            )
+            let ret = ffi::PyObject_SetAttr(self.as_ptr(), attr_name, value);
+            ffi::Py_DECREF(attr_name);
+            ffi::Py_XDECREF(value);
```

Comment:
e.g. here there's no need for `XDECREF` instead of `DECREF` because it's got to be a non-null pointer (it came via `to_object` call).